packages/lib/src/services/app-hosting/__tests__/app-replay-key.test.ts

```
@@ -0,0 +1,138 @@
+
+const SECRET = 'x'.repeat(32);
+const OTHER_SECRET = 'y'.repeat(32);
+
+const key = (flyAppName: string, secret = SECRET) =>
+  derivePublishedAppReplayKey({ flyAppName, secret });
+
+describe('derivePublishedAppReplayKey — deterministic, per app, per secret', () => {
+  assert({
+    given: 'the same app name and secret twice',
+    should: 'derive the same key, so the router and the guest agree without exchanging anything',
+    actual: key('pgs-app-abc') === key('pgs-app-abc'),
```

Biome (2.5.7)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)